src/tiles.js

```diff
@@ -23,6 +23,9 @@ import {
   setObjectValues,
 } from './utils'
 
+// register dynamic import for compression codec
+addCodec('zlib', () => import('numcodecs/zlib').then(m => m.default))
```

Comment:
This could equivalently be if code-splitting the codec and dynamically loading at runtime isn't desired.

```suggestion
import Zlib from 'numcodecs';
addCodec('zlib', () => Zlib);
```